### ISSUE TITLE
Delegating "Ecto.Migration.Schema" more to adapters

### ISSUE THREAD

**Issue by whatyouhide** [elixir-ecto/ecto_sql#533]
### Elixir version

1.15.0

### Database and Version

AWS Redshift

### Ecto Versions

main

### Database Adapter and Versions (postgrex, myxql, etc)

postgrex, whatever version

### Current behavior

Right now, things can get ugly when running migrations because Ecto takes a few decisions around where and how to keep track of migrations. There are ways to configure *parts* of it (like changing the name of the table from `schema_migrations` to something else), but a few things are sort of hard-coded. For example, I recently (https://github.com/elixir-ecto/ecto_sql/pull/531) had to add an option to avoid doing `CAST(version AS int)` in migration-related queries, because that breaks on some databases.

Some databases, such as [AWS Redshift](https://aws.amazon.com/redshift/), have even more restrictions. For example, Redshift doesn't really support a bunch of data types and is picky about stuff.

### Expected behavior

I'd like to figure out how we can improve the situation here. For example, we could have the adapter have callbacks that suggest Ecto how to store migration information.

I posted this in an issue because I do think this can create issues against some databases, but I'm happy to move this to the mailing list if preferred.

**Comment by whatyouhide** [comment]
@josevalim fair enough! One issue that we are having with Redshift is that Redshift does not support timestamps with precision ([docs](https://docs.aws.amazon.com/redshift/latest/dg/r_Datetime_types.html)), and Ecto stores the `inserted_at` column of `schema_migrations` as a `:utc_timestamp`, which gets translated to `timestamp(0)` in the Postgres adapter.

Now, I can hear you already: Postgrex and the Postgres Ecto adapters are not meant to work with Redshift 😄 So maybe the solution is just a different adapters that delegates to the Ecto one. This is what was done in https://github.com/100Starlings/redshift_ecto, as a matter of fact.

So maybe I was too quick to open this issue. Let me close it for now, and if I get a better argument on my hands, I'll re-open 🙃

**Comment by josevalim** [comment]
FWIW, I am fine with making the time stamp customizable because it will also impact outside of migrations.
